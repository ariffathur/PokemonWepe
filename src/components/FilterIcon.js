import React from "react";
import PropTypes from "prop-types";
import { TouchableOpacity } from "react-native";
import { FontAwesome } from "@expo/vector-icons";

const FilterIcon = (props) => {
  return (
    <TouchableOpacity style={props.style} onPress={props.onPress}>
      <FontAwesome name="filter" size={props.size} color={props.color} />
    </TouchableOpacity>
  );
};

FilterIcon.propsTypes = {
  style: PropTypes.object,
  onPress: PropTypes.func,
  size: PropTypes.number,
  color: PropTypes.string,
};

FilterIcon.defaultProps = {
  onPress: null,
  style: null,
  size: 24,
  color: "#ffff",
};

export default FilterIcon;
