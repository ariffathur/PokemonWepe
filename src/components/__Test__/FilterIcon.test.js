import React from "react";
import renderer from "react-test-renderer";

import FilterIcon from "../FilterIcon";

describe("<FilterIcon />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<FilterIcon />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
