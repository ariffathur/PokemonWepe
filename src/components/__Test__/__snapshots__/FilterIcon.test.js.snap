// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterIcon /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
